projects/importing/attribute.py

```diff
@@ -165,66 +231,55 @@ def _update_sections(self, data):
                     action_str = "Created"
                 else:
                     action_str = "Updated" if overwrite else "Already exists, skipping"
 
-                logger.info("{} {}".format(action_str, section))
+                logger.info(f"{action_str} {section}")
 
-    def _replace_attribute_section_links(self, data):
+    def _replace_attribute_section_links(self, rows):
         logger.info("\nReplacing attribute section links...")
 
         ProjectPhaseSectionAttribute.objects.filter(
             section__phase__project_type=self.project_type
         ).delete()
 
-        counter = Counter()
-        for datum in data[1:]:
-            identifier = self._get_datum_identifier(datum)
-            try:
-                attribute = Attribute.objects.get(identifier=identifier)
-            except Attribute.DoesNotExist:
-                logger.warning('Attribute "{}" does not exist.'.format(identifier))
-                continue
+        for phase in ProjectPhase.objects.filter(project_type=self.project_type):
 
-            for phase_num in [0, 1, 2]:
-                phase = ProjectPhase.objects.get(
-                    project_type=self.project_type, index=phase_num
-                )
+            # Index for attribute within a phase section
+            counter = Counter()
 
-                section_name = datum[SECTION_COLUMNS[phase_num]]
-                if not section_name or not isinstance(section_name, str):
+            for row in rows:
+                identifier = self._get_attribute_identifier(row)
+                attribute = Attribute.objects.get(identifier=identifier)
+
+                if phase.index not in self._get_attribute_input_phases(row):
                     # Attribute doesn't appear in this phase
                     continue
 
-                section_name = section_name.strip()
-                try:
-                    section = ProjectPhaseSection.objects.get(
-                        phase=phase, name=section_name
-                    )
-                except ProjectPhaseSection.DoesNotExist:
-                    logger.warning(
-                        'Section "{}" in phase {} does not exist.'.format(
-                            section_name, phase
-                        )
-                    )
+                section_phase_name = row[self.column_index[PHASE_SECTION_NAME]].strip()
+
+                section = ProjectPhaseSection.objects.get(
+                    phase=phase, name=section_phase_name
+                )
 
                 is_generated = False
 
-                master_data_description = datum[1].strip()
-                if (
-                    "muodostuu" in master_data_description
-                    and "perusteella" in master_data_description
-                ):
-                    is_generated = True
+                # TODO Column for setting ProjectPhaseSectionAttribute.generated is missing
+                # master_data_description = row[1].strip()
```

Comment:
I don't really like commented out code unless it is in an example docstring or something like that. Can this be removed?

If you have an idea of how it could be done, create a branch and store it there for instance. This is, for me seen as dead code.